### PR TITLE
fixes issue #6349 - check if posix is available before using it

### PR DESCRIPTION
**Pull request by bjoern86**
check if posix is available before using it, if not available set name in Array to NULL

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by bjoern86** [comment]
the following commits fix #6349

commit c26f7924c870a869839f055d7a8b6770a74ce44b
commit 0bb206eaaad934cd095137306d30935c3d4da8df
commit eedaba0058dc8ba0ec076878f0c9f8e9ce71d980

who can merge them?

**Comment by samus-aran** [comment]
Hey @bjoern86 There are two other checks that would need to the same function check. Can you also include them into this PR?

modules/UpgradeWizard/SystemCheck.php
modules/UpgradeWizard/uw_utils.php

We would really appreciate that! :+1:

File: modules/UpgradeWizard/uw_ajax.php [hunk 1]
```diff
@@ -892,8 +892,8 @@ function systemCheckJsonCheckFiles($persistence)
                 // don't warn yet - we're going to use this to check against replacement files
                 $filesNotWritable[$i] = $file;
                 $filesNWPerms[$i] = substr(sprintf('%o', fileperms($file)), -4);
-                $owner = posix_getpwuid(fileowner($file));
-                $group = posix_getgrgid(filegroup($file));
+                $owner = (function_exists('posix_getpwuid') ? posix_getpwuid(fileowner($file)) : array('name' => 'NULL'));
```
Comment (gymad) [inline modules/UpgradeWizard/uw_ajax.php]: is `NULL` string enough information to figure out what happends here? Please add a log notes to describe the problem at least.

**Comment by samus-aran** [comment]
Hey @bjoern86 Are you able to assist with completing this PR? It would be much appreciated!

**Pull request closed without merging**